src/index.js

```diff
@@ -1,15 +1,43 @@
-import React from "react";
-import ReactDOM from "react-dom";
+import React, { StrictMode } from "react";
+import * as ReactDOMClient from "react-dom/client";
 import "./index.css";
 import App from "./App";
 import { makeServer } from "./server";
 
+import { BrowserRouter as Router } from "react-router-dom";
+
+import {
+  VideoProvider,
+  AuthProvider,
+  LikeProvider,
+  WatchLaterProvider,
+  HistoryProvider,
+  PlaylistProvider,
+  SinglePlaylistProvider
+} from "./context/index";
 // Call make Server
 makeServer();
 
-ReactDOM.render(
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>,
-  document.getElementById("root")
-);
+const root = ReactDOMClient.createRoot(document.getElementById("root"));
+root.render(
+  <StrictMode>
+    <Router>
+      <AuthProvider>
+        <LikeProvider>
+          <VideoProvider>
+            <WatchLaterProvider>
+              <HistoryProvider>
+                <PlaylistProvider>
+                <SinglePlaylistProvider>
+
+                  <App />
+                </SinglePlaylistProvider>
+                </PlaylistProvider>
+              </HistoryProvider>
+            </WatchLaterProvider>
+          </VideoProvider>
+        </LikeProvider>
```

Comment:
you may use single provider instead of using multiple provider. like main context and auth context.